### PR TITLE
145458 List: Support scroll by page

### LINKED ISSUE #145458: List: Support scroll by page
I searched for the request in subject, but didn't find it. I saw numerous requests to make scrolling page by page in editors, and thanfully we got that option with the "editor.scrollbar.scrollByPage". However, the scrolling in almost every other situation is absolute, i.e. you get scroll to the place where you clicked. AFAIK, the only other place in VSCode where we have scrolling page by page is the extension web pages displayed in the editor, which works like that even if "editor.scrollbar.scrollByPage": false.

I would like if the mentioned scrollByPage setting is global for all places that have scroll bars, like Explorer/Search/Extensions panels. Or, you could add another setting which will apply to these other places independently of the editor setting.

### PR DESCRIPTION
**Pull request by Balastrong**
Option scrollByPage is now available for list, to enable the feature on every scrollbar.

This PR fixes #145458 

(I ended up copy-pasting more code than I initially expected, please let me know if there's something to remove or everything is needed to cover all cases)

### CODE REVIEW

**Review by Qasem-h — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by joaomoreno — APPROVED** [review]

**Comment by joaomoreno** [comment]
Sorry for the delay reviewing this. Pushed minor changes as a lot of the code wasn't really necessary. Waiting for a second approval to merge this.

Many thanks and great work! 🍻

**Comment by Balastrong** [comment]
Thank you for the review :D

**Review by lszomoru — APPROVED** [review]

**Pull request merged**
